lib/auth/apiserver.go

```diff
@@ -2494,6 +2498,12 @@ func (s *APIServer) getServerID(r *http.Request) (string, error) {
 	return strings.TrimSuffix(role.Username, "."+clusterName), nil
 }
 
+// isReservedOrigin returns true if the given origin is reserved for system use
+// and must not be accepted within a user resource.
+func isReservedOrigin(origin types.OriginValue) bool {
```

Comment:
If we change the label to be namespaced as I suggested above, would it make sense to just prevent it to be set by users completely instead of just restricting allowed values?

Reply:
Has to be brought up with @klizhentas as it would be a modification of the RFD design.